### PR TITLE
Turned exercise 060 into practice

### PR DESCRIPTION
**Pull request by maha-sachin**
Turned Exercise 060 into Practice 060

Added implementation for the `hbal_tree_nodes` function in the solution for Exercise 060:
- Implemented the `hbal_tree_nodes` function to generate all height-balanced binary trees with a given number of nodes.
- Included detailed test cases to ensure correctness and handled boundary cases appropriately:
  - Test cases for `hbal_tree_nodes`:
    - Checked generation of height-balanced trees for various heights (0, 1, 2).
    
  - Test cases for `max_nodes` and `min_nodes`:
    - Verified the calculation of the maximum and minimum number of nodes based on the height of the tree.
    - Tested with different heights to cover the range of possible nodes.

Implemented the `max_nodes` and `min_nodes` functions to calculate the maximum and minimum number of nodes in a height-balanced tree, respectively.

### CODE REVIEW

File: practice/060/ex.ml
Comment (cuihtlauac) [inline practice/060/ex.ml:1]: Add more tests here too, possible use trees from other exercises

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
The function `max_nodes` is provided in the description of the task, therefore it should be part of file `work/impl.ml` and preferably as provided initially

**Comment by maha-sachin** [comment]
Hi @cuihtlauac , I've committed the change to include the max_nodes function in the work/impl.ml file as provided initially.

**Review by cuihtlauac — APPROVED** [review]
Thanks @maha-sachin, this is nice

**Comment by maha-sachin** [comment]
> Thanks @maha-sachin, this is nice

@cuihtlauac Thank you. I'm eager to take on a new task and contribute further to the project. Could you please assign me to another issue or let me know if there's anything else I can assist with?

**Pull request merged**